main/acle.md

```diff
@@ -4932,6 +5044,1381 @@ assertions, which may speed up access to aligned data (and will fault
+of [[AAPCS64]](#AAPCS64). The mapping of this PCS concept to C and C++
+depends on whether the function is [shared-ZA](#shared-za) or
+[private-ZA](#private-za):
+
+*   If the function is [shared-ZA](#shared-za), the attribute guarantees
+    that the contents of the shared [ZA state](#za-state) on return from
+    the function are the same as the contents of the shared ZA state on
+    entry to the function. Either the function does not change the ZA
+    state at all, or the function undoes any changes to the ZA state
+    before returning.
+
+    Note that C and C++ calls from shared-ZA functions to private-ZA
+    functions are defined to preserve ZA state. Such calls do not affect
+    whether a function can be marked [`arm_preserves_za`](#arm_preserves_za).
+
+*   If the function is [private-ZA](#private-za), the attribute guarantees
+    that a call to the function does not directly or indirectly involve
+    any of the following:
+
+    *   a call to an [`arm_new_za`](#arm_new_za) function
+
+    *   a call to `setjmp`
+
+    *   a call to [`__arm_za_disable`](#__arm_za_disable)
+
+    *   a call to a non-C/C++ function that “commits a lazy save”, in the
+        sense of the [[APCS64]](#AAPCS64)
+
+    *   an inline asm that commits a lazy save
+
+    *   the catching of an exception
+
+    The platform may place additional requirements as well.
+
+In both cases, the onus is on the definition of the function to honor
+the guarantee that is being made. The attribute does not direct the
+compiler to do anything to honor the guarantee.
+
+If a function with an `arm_preserves_za` type does not preserve ZA,
+the behavior is undefined. (There is an analogy with functions
+that are declared `noreturn` but do in fact return, and to functions
+that are declared `const` but do in fact change memory.)
+
+The attribute is intended to be useful for functions at API boundaries,
+where the compiler might not be able to “see” the definition of the
+function being called. As the description above implies, attaching
+`arm_preserved_za` to a [private-ZA](#private-za) function is quite
+a low-level feature, but it is useful for
+[streaming-compatible versions of standard routines](#streaming-compatible-versions-of-standard-routines)
+and could be useful for things like vector math routines.
+
+Function types with this attribute implicitly convert to function types
+that do not have the attribute. The reverse is not true. For example:
+
+``` c
+  #define ATTR __attribute__((arm_preserves_za))
+
+  ATTR void (*ptr1)(void);
+  void (*ptr2)(void);
+  ATTR void f1(void);
+  void f2(void);
+
+  void code() {
+    ptr1 = ptr2;  // Ill-formed
+    ptr1 = f1;    // OK
+    ptr1 = f2;    // Ill-formed
+
+    ptr2 = ptr1;  // OK
+    ptr2 = f1;    // OK
+    ptr2 = f2;    // OK
+  }
+```
+
+## SME functions
+
+The functions in this section are defined by the header file
+[`<arm_sme.h>`](#arm_sme.h).  Except where noted otherwise,
+the functions do not have linkage.
```

Comment:
When they do have linkage, what do you need to link them to? Is that something provided by the toolchain and transparent to the user? Or do they need to proactively link their object files against `-lWhatever`?

Reply:
A function “has linkage” if it can appear in the symbol table of an object file.  I can/should add that to the “Terms and abbreviations” section.